crates/noirc_frontend/src/hir_def/types.rs

```diff
@@ -1212,6 +1222,12 @@ impl Type {
                     .expect("Cannot have variable sized strings as a parameter to main");
                 AbiType::String { length: size }
             }
+            Type::FmtString(size, _) => {
+                let size = size
+                    .evaluate_to_u64()
+                    .expect("Cannot have variable sized strings as a parameter to main");
+                AbiType::String { length: size }
+            }
```

Comment:
We should probably disallow all fmt strings from being used in the abi. If the user wanted a normal string they should have written one. There's no way to encode the environment info of a fmt string here anyway.

Reply:
Yeah you write I disallowed it

Reply:
Yeah you are correct I have disallowed it